figo/figo.py

```diff
@@ -510,12 +510,10 @@ def remove_account(self, account_or_account_id):
         :Parameters:
          - `account_or_account_id` - account to be removed or its ID
         """
-        if isinstance(account_or_account_id, STRING_TYPES):
-            self._request_with_exception("/rest/accounts/%s" % account_or_account_id,
-                                           method="DELETE")
-        else:
-            self._request_with_exception("/rest/accounts/%s" % account_or_account_id.account_id,
-                                           method="DELETE")
```

Comment:
This is an obvious code smell and the way you fixed it is definitely an improvement. On the other hand I think that the real problem is, that the interface is crappy, because it allows you to do sloppy typing which many time results in problems. I think this (and possibly many other methods) should just allow the user to pass an id and not an id *or* an object. This will however be a braking change to the API, because users might rely on this behaviour. 
Another idea would be to implement the `__str__` special methods of the API-objects in a way that it returns the object's id.